app/models/Level.coffee

```diff
@@ -293,6 +293,13 @@ module.exports = class Level extends CocoModel
   isType: (types...) ->
     return @get('type', true) in types
 
+  getSolutionFor: (language) ->
+    return unless hero = _.find (@get("thangs") ? []), id: 'Hero Placeholder'
+    return unless plan = _.find(hero.components ? [], (x) -> x?.config?.programmableMethods?.plan)?.config.programmableMethods.plan
+    return unless solution = plan.solutions.find((s) -> s.language == language && s.succeeds == true)
+    return unless solution.source
+    return _.template(solution?.source)(utils.i18n(plan, 'context'))
```

Comment:
Should handle errors here, like we did recently [here](https://github.com/codecombat/codecombat-server/blob/master/server/models/Level.coffee#L95-L98), since we often end up with broken templates.